tools/upgradetool/src/main/java/org/openhab/core/tools/internal/ItemUnitToMetadataUpgrader.java

```diff
@@ -121,7 +101,7 @@ public void itemCopyUnitToMetadata() {
                     logger.debug("{}: Already contains a 'unit' metadata, skipping it", itemName);
                 } else {
                     String unit = null;
-                    if (!noLink) {
+                    if (linkStorage != null && thingStorage != null) {
```

Comment:
Is the test on `noLink` no more needed?

Reply:
The code is equivalent, except that this one helps avoid compiler warnings about possible unchecked null value.